apm-sniffer/apm-sdk-plugin/vertx-plugins/vertx-core-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/vertx3/HttpClientRequestImplEndInterceptor.java

```diff
@@ -40,11 +40,17 @@ public void onConstruct(EnhancedInstance objInst, Object[] allArguments) {
         String host;
         int port;
         if (allArguments[3] instanceof Integer) {
+            //ver. 3.0.x - 3.3.x
```

Comment:
This is an HTTP interceptor, I think `instanceof` is acceptable. Only remind you, using witness class to different versions is better. Because it has no payload at the runtime, even in the JDK8 `instanceof` is faster, but at least nothing is faster :)

Reply:
Wouldn't that mean creating three `HttpClientRequestImplEndInterceptor` classes? One for each version range

Reply:
Yeap. If they have a right witness class in each version range.

Reply:
You could put three sub interceptors, as the difference is onlu about onConstruct. WDYT?